src/nn/ops/resample2d.ts

```diff
@@ -1,34 +1,40 @@
 import * as tf from '@tensorflow/tfjs-core';
 
-import {MLInterpolationMode, MLResampleOptions} from '../graph_builder';
+import {MLInterpolationMode, MLResample2dOptions} from '../graph_builder';
 import {MLOperand} from '../operand';
 import {SingleOutputOperation} from '../operation';
 import * as utils from '../utils';
 
-export class Resample extends SingleOutputOperation {
+export class Resample2d extends SingleOutputOperation {
   private input_: MLOperand;
   private mode_: MLInterpolationMode = MLInterpolationMode['nearest-neighbor'];
-  private scales_: [number, number, number, number];
-  private sizes_: [number, number, number, number];
+  private scales_: [number, number] = [1.0, 1.0];
+  private sizes_: [number, number];
+  private axes_: [number, number] = [2, 3];
 
-  constructor(input: MLOperand, options: MLResampleOptions = {}) {
+  constructor(input: MLOperand, options: MLResample2dOptions = {}) {
     super(input.builder);
     utils.validateOperand(input);
     this.input_ = input;
     if (options.scales !== undefined) {
       const array = options.scales;
       utils.assert(
           array instanceof Array && array.every(v => typeof v === 'number') &&
-              array.length === 4,
+              array.length === 2,
           'The scales parameter is invalid.');
       this.scales_ = options.scales;
     }
     if (options.sizes !== undefined) {
       utils.assert(
-          utils.isIntegerArray(options.sizes) && options.sizes.length === 4,
+          utils.isIntegerArray(options.sizes) && options.sizes.length === 2,
           'The sizes parameter is invalid.');
       this.sizes_ = options.sizes;
-      this.scales_ = undefined;
+    }
+    if (options.axes !== undefined) {
+      utils.assert(
+          utils.isIntegerArray(options.axes) && options.axes.length === 2,
+          'The axes parameter is invalid.');
```

Comment:
Please also check the valid axes values here.

Reply:
Thanks @huningxin, I will add it in next commit.